kaizhilam/ProjectCreature
# new state, inv fully working, reorganising manager stuff

* Changed QuestController.cs into QuestManager.cs and put inside Manager folder
* Changed CursorController.cs into CursorManager.cs and put inside Manager folder
* Moved Quest.cs into Player Folder
* Deleted GameController folder
* Renamed GameController in hierarchy to GameManager
* Deleted SoundController object in hierarchy and moved script and audiosource to GameManager object
* Created GameManager.cs inside Manager folder (singleton) responsible for game over stuff and broadcasting large events
* Added many new functions to InventoryManager
* If item you pickup is already in the inventory and can stack (capacity>1) it will stack together
* Text now correctly shows how many items are in each stack
* Created hotbar (needs UI rework)
* If hotbar is full, items will start to be added to inventory instead
* Added ChaseState.cs to State folder
* Freeing myself from the hell that is Kai's Debug.Log() running each tick ;(
* Created WeaponFactory.cs which will return weapon if you input string through GetWeapon() function


How to test this pull request:
Pick up all 7 weapons - weapons should stack to maximum size of 2, if at capacity, will be inserted at next available slot, if hotbar full, it will add to inventory
Check that sounds still work

PLEASE ONLY MERGE THIS ONCE QUAN HAS A NEW PULL REQUEST MERGED, pls review tho x
